### ISSUE TITLE
APIs should return 400s if the provided input is not parseable

### ISSUE THREAD

**Issue by b1bart** [EOSIO/eos#8970]
## Overview

In #8899 with commits 954f4bc35 and e03a5d2d5 the HTTP API endpoint added no longer results in a `500` when the parameters passed in as the `POST` body are not parseable as the expected data structure.  Instead, those failures are caught and respond with a `400` to indicate to the user that there is something wrong with their request and not falsely implicate the server application.  

For that PR, we confined the scope of the change to the features added in the PR, however, there are many other APIs that have this same basic issue which should be corrected.

Simply put, if the failure is caused by malformed input by the user then the HTTP error code should be `400`

All existing HTTP APIs should be audited and corrected.

**Comment by allenhan2** [comment]
This one has been fixed
